### ISSUE TITLE
Primary ctor parameter writes should be assumed necessary even if not read later in the method

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#68904]
**Version Used**: VS 17.7.0 Preview 2.0

**Steps to Reproduce**:

```cs
class MyClass(int value)
{
    public int Get()
    {
        var x = value;
        value++; // IDE0059: unnecessary assignment of a value to 'value'
        return x;
    }
}
```

**Expected Behavior**: No diagnostic. Behavior should be similar to if a field 'int value' were explicitly declared in 'MyClass'.

**Actual Behavior**: IDE0059 is reported

Relates to https://github.com/dotnet/roslyn/issues/65697

**Comment by RikkiGibson** [comment]
Ping @arkalyanms @dotnet/roslyn-ide for triage

**Comment by arkalyanms** [comment]
Starting with Cyrus for first look.

**Comment by mavasani** [comment]
Duplicate of #69643. Verified that this was fixed with #69748
